Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py

```diff
@@ -0,0 +1,563 @@
+            method='POST',
+            url_suffix='token',
+            data={'grant_type': 'client_credentials'},
+            auth=HTTPBasicAuth(self.public_key, self.private_key),
+            timeout=self._polling_timeout
+        )
+
+        auth_token = response.get('access_token')
+        expires_in = response.get('expires_in')
+        epoch_expiration_time = self.parse_access_token_expiration_time(expires_in)
+
+        demisto.setIntegrationContext(
+            {
+                'auth_token': auth_token,
+                'expiration_time': epoch_expiration_time
+            }
+        )
+
+        return auth_token
+
+    def get_access_token(self) -> str:
+        """Returns the current valid authentication token for the feed.
+
+        Returns:
+            str. Authentication token.
+        """
+        last_token_fetched_expiration_time = demisto.getIntegrationContext().get('expiration_time')
+        current_time = int(datetime.now().strftime('%s'))
+
+        if last_token_fetched_expiration_time and last_token_fetched_expiration_time > current_time:
+            auth_token = demisto.getIntegrationContext().get('auth_token')
+        else:
+            auth_token = self.fetch_new_access_token()
+
+        return auth_token
+
+    def fetch_all_indicators_from_api(self, limit: int) -> Tuple[List, Dict, Dict]:
+        """Collects raw data of indicators and their relationships from the feed.
+
+        Args:
+            limit (int): Amount of indicators to fetch. -1 means no limit.
+
+        Returns:
+            Tuple[List, Dict, Dict].
+                raw_indicators - List of STIX 2.1 indicators objects.
+                relationships - Dict of `id: STIX 2.1 relationship object`.
+                stix_entities - Dict of `id: STIX 2.1 entity object`.
+        """
+        raw_indicators = list()  # type: List
+        relationships = dict()  # type: Dict
+        stix_entities = dict()  # type: Dict
+
+        headers = {
+            'Accept': 'application/vnd.oasis.stix+json; version=2.1',
+            'X-App-Name': 'content.xsoar.cortex.paloaltonetworks.v1.0',
+            'Authorization': f'Bearer {self.get_access_token()}'
+        }
+
+        if limit == -1:
+            query_url = '/collections/indicators/objects?length=1000'
+        else:
+            query_url = f'/collections/indicators/objects?length={min(limit, 1000)}'
+
+        while True:
+            response = self._http_request(
+                method='GET',
+                url_suffix=query_url,
+                headers=headers,
+                timeout=self._polling_timeout,
+                resp_type='response'
+            )
+
+            if response.status_code == 204:
+                demisto.info(f'{INTEGRATION_NAME} info - '
+                             f'API Status Code: {response.status_code} No Content Available for this timeframe.')
+                return [], {}, {}
+
+            if response.status_code != 200:
```

Comment:
shouldn't this return_error of some kind? if not how do we alert the user about a possible error? - same for `fetch_all_reports_from_api`